autowiring/CoreObjectDescriptor.h

```diff
@@ -36,7 +37,11 @@ struct CoreObjectDescriptor {
     pFilter(autowiring::fast_pointer_cast<ExceptionFilter>(value)),
     pBoltBase(autowiring::fast_pointer_cast<BoltBase>(value)),
     receivesEvents(
-      [this]{
+      [this, value]{
+        // Because we manually added AutowiringEvents to the JunctionBoxManager, check here also
+        if (autowiring::fast_pointer_cast<AutowiringEvents>(value))
```

Comment:
We've had problems where `AutowiringEvents` isn't added to the registry. A complete fix would guarantee its added to the registry so we can eliminate this quick fix, as well as [this hack](https://github.com/leapmotion/autowiring/blob/7c9312061fbcae4b81b93f27f0af1659d32c4969/src/autowiring/JunctionBoxManager.cpp#L19).